### PR TITLE
remove unwrap from s3 storage backend and simplify obj list code

### PR DESCRIPTION
**Pull request by houqp**

### CODE REVIEW

File: rust/src/storage/s3.rs [hunk 1]
```diff
@@ -13,6 +12,47 @@ use tokio::io::AsyncReadExt;
 
 use super::{parse_uri, ObjectMeta, StorageBackend, StorageError};
 
+enum ContinuationToken {
```
Comment (rtyler) [inline rust/src/storage/s3.rs:15]: I see what this code is doing, but I don't fully understand why. Would you mind adding some rustdoc @houqp

File: rust/src/storage/s3.rs [hunk 2]
```diff
@@ -13,6 +12,47 @@ use tokio::io::AsyncReadExt;
...
+    Value(Option<String>),
+    End,
+}
+
+fn err_s3_obj_missing_key() -> StorageError {
+    StorageError::S3Generic("S3 Object missing key attribute".to_string())
+}
+
+fn err_s3_obj_missing_modified() -> StorageError {
```
[the remark this answers is not in the document]
Comment (fvaleye) [reply rust/src/storage/s3.rs:24]: I share the same feeling, it might be a good thing to use it locally and refactor it after if needed?

**Review by fvaleye — APPROVED** [review]

File: rust/src/storage/s3.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (houqp) [reply rust/src/storage/s3.rs]: yeah, they were used in multiple places initially, but some where along the way, i refactored those invocations into `parse_obj_last_modified_time` and `impl TryFrom<rusoto_s3::Object> for ObjectMeta`, which turned them into single use helper functions :P I have pushed a new commit to inline them.

File: rust/src/storage/s3.rs (hunk 1 above)
Comment (houqp) [reply rust/src/storage/s3.rs]: I have moved this enum into the list_objs function so it only stays in the scope where it's being used. I have also added some doc string to expand on how it's being used in the object list code.

**Review by fvaleye — APPROVED** [review]

**Review by rtyler — APPROVED** [review]
Thanks for the explanations @houqp , makes more sense to me now :smile_cat:

**Pull request merged**